### ISSUE TITLE
Support for external deps

### ISSUE THREAD

**Issue by xtuc** [Rich-Harris/packd#27]
Following the discussion from https://github.com/babel/website/issues/1302.

It would be great to allow externals for both Browserify and Rollup. What do you think @Rich-Harris?

PS: i'm currently working on it.

**Comment by xtuc** [comment]
Imagine I want to dedupe `babel-types` (babel-plugin-ivi-jsx is requiring it):

```html
<script type="text/javascript" src="http://localhost:9000/babel-types"></script>
<script type="text/javascript" src="http://localhost:9000/babel-plugin-ivi-jsx"></script>
```

When I declare `babel-types` as an external dependency, browserify fails at runtime with `Cannot find module 'babel-types'`. In the page `babel-types` is a available as `window.babelTypes`. I don't know how to tell browserify to use it.

Maybe we could generate a fake package called `babel-types` somehow which just contains a alias statement: `module.exports = window.babelTypes` but sounds hacky and fragile.

Do you have an idea?

**Comment by Daniel15** [comment]
What about allowing externals to be passed as querystring parameters? For example `/babel-plugin-ivi-jsx?externals[babel-types]=babelTypes` could do the equivalent of doing this in a Webpack config:
```
externals: {
  'babel-types': 'BabelTypes',
}
```

**Comment by xtuc** [comment]
Yeah, configuration is only possible through query parameters (because of the CDN).

I planned something like this: `/babel-plugin-ivi-jsx?externals=babel-types,babel-core`.

I don't get how browserify is working with externals, they are not passed as arguments as with Rollup. Maybe @Rich-Harris could help us here.

**Comment by Rich-Harris** [comment]
This seems like a good. Afraid I have no idea how to make it work with Browserify! 😬

**Comment by ceremcem** [comment]
Our project is using browserify, you may want to take a look at the [`gulpfile`](https://github.com/aktos-io/scada.js/blob/2c423e338686cd71f47350e1e911b492848950a9/gulpfile.ls). My two cents...

**Comment by xtuc** [comment]
We need this for the Babel REPL. Since Browserify is only used as a fallback, I think it's fine to support externals only when you bundle with Rollup (and your module supports it).

**Comment by xtuc** [comment]
Well now we have a PR https://github.com/Rich-Harris/packd/pull/37 but I don't get it, it can't work. If someone has the time to look at it :pray:
